js/queue/queue-data-structure.js

```diff
@@ -10,17 +10,51 @@ class QueueDataStructure {
 
   canEnqueue() {
     // ... your code goes here
+    if (this.queueControl <= this.MAX_SIZE) {
+      return true
+    } else {
+      return false
+    }
   }
 
   isEmpty() {
     // ... your code goes here
+    if (this.queueControl.length == 0) {
+      return true
+
+    } else {
+      return false
+    }
   }
 
   enqueue(item) {
     // ... your code goes here
+    this.queueControl.push(item)
+
+
+    if (this.queueControl.length == this.MAX_SIZE) {
+      return "Stack Overflow"
+    }
+
+    return this.queueControl
+
+
+
```

Comment:
Puedes dejar huecos pero que tengan algo de sentido. Por ejemplo aquí dejo huecos alrededor del if para que se lea bien.
```suggestion
    this.queueControl.push(item)

    if (this.queueControl.length == this.MAX_SIZE) {
      return "Stack Overflow"
    }

    return this.queueControl
```